src/actions/public/L2/estimateFees.test.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * The first 2 test cases are good documentation of how to use this library
+ */
+import { l2StandardBridgeABI, l2StandardBridgeAddress, optimistABI, optimistAddress } from '@eth-optimism/contracts-ts'
+import { createPublicClient, http, parseEther, parseGwei } from 'viem'
+import { optimism } from 'viem/chains'
+import { formatEther } from 'viem/utils'
+import { beforeEach, expect, test, vi } from 'vitest'
+import { estimateFees } from './estimateFees.js'
+import { estimateL1Fee } from './estimateL1Fee.js'
+
+vi.mock('viem', async () => {
+  const _viem = (await vi.importActual('viem')) as any
+  return {
+    ..._viem,
+    // no way to get historical gas price
+    createPublicClient: (...args: [any]) => {
+      const client = _viem.createPublicClient(...args)
+      client.getGasPrice = async () => parseGwei('0.00000042')
+      return client
+    },
+  }
+})
+
+// using this optimist https://optimistic.etherscan.io/tx/0xaa291efba7ea40b0742e5ff84a1e7831a2eb6c2fc35001fa03ba80fd3b609dc9
+const blockNumber = BigInt(107028270)
+const optimistOwnerAddress = '0x77194aa25a06f932c10c0f25090f3046af2c85a6' as const
+const functionDataBurn = {
+  functionName: 'burn',
+  // this is an erc721 abi
+  abi: optimistABI,
+  args: [BigInt(optimistOwnerAddress)],
+  account: optimistOwnerAddress,
+  to: optimistAddress[10],
+  chainId: 10,
+} as const
+const functionDataBurnWithPriorityFees = {
+  ...functionDataBurn,
+  maxFeePerGas: parseGwei('2'),
+  maxPriorityFeePerGas: parseGwei('2'),
+} as const
+// This tx
+// https://optimistic.etherscan.io/tx/0xe6f3719be7327a991b9cb562ebf8d979cbca72bbdb2775f55a18274f4d0c9bbf
+const functionDataWithdraw = {
+  abi: l2StandardBridgeABI,
+  functionName: 'withdraw',
+  value: BigInt(parseEther('0.00000001')),
+  account: '0x6387a88a199120aD52Dd9742C7430847d3cB2CD4',
+  // currently a bug is making chain id 10 not exist
+  to: l2StandardBridgeAddress[420],
+  chainId: 10,
+  args: [
+    // l2 token address
+    '0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000',
+    // amount
+    BigInt(parseEther('0.00000001')),
+    // l1 gas
+    0,
+    // extra data
+    '0x0',
+  ],
+  maxFeePerGas: parseGwei('.2'),
+  maxPriorityFeePerGas: parseGwei('.1'),
+} as const
+
+const viemClient = createPublicClient({
+  chain: optimism,
+  transport: http(process.env.VITE_L2_RPC_URL ?? 'https://mainnet.optimism.io'),
```

Comment:
will this only work with optimism? If so, I think you don't want to allow VITE_L2_RPC_URL, if it'll work with others, you should use https://github.com/base-org/op-viem/blob/main/src/actions/public/L2/getWithdrawalMessages.test.ts#L2

Reply:
Should append the chain Id to the env variable name imo it only works with op

Reply:
> Should append the chain Id to the env variable name imo

sorry can you say what you mean? Like VITE_8453_RPC_URL?

Reply:
hmm I think would break precedent with viem? sees ideally there's one rollup fork and one L1 fork?

Reply:
Yea we can keep iterating on it. Just for historical context this test isn't meant to be what gets upstreamed from viem it's just the test that happened to get written when this originally landed in the optimism monorepo. We should definitely replace it with a higher quality test.

A project we are kicking off now is the ability to test something like this using a devnet like anvil which this (and all of op-viem) would be a good use case for dogfooding.  One of the requirements of this project is to be able to test things like l1 gas fee against a devnet